recipes/libpnetcdf/meta.yaml

```diff
@@ -0,0 +1,76 @@
+{% set version = "1.12.3" %}
+{% set build = 0 %}
+
+# recipe-lint fails if mpi is undefined
+{% set mpi = mpi or 'mpich' %}
+{% if mpi == "mpich" %}
+# prioritize mpich via build number
+{% set build = build + 100 %}
+{% endif %}
+
+package:
+  name: libpnetcdf
+  version: {{ version }}
+
+source:
+  url: https://parallel-netcdf.github.io/Release/pnetcdf-{{ version }}.tar.gz
+  sha256: 439e359d09bb93d0e58a6e3f928f39c2eae965b6c97f64e67cd42220d6034f77
+
+build:
+  number: {{ build }}
+  skip: True  # [win]
+  {% set mpi_prefix = "mpi_" + mpi %}
+  # add build string so packages can depend on mpi variants dependencies:
+  # `PKG_NAME * mpi_mpich_*` for mpich
+  # `PKG_NAME * mpi_*` for any mpi
+  string: {{ mpi_prefix }}_h{{ PKG_HASH }}_{{ build }}
+
+  run_exports:
+    - {{ pin_subpackage('libpnetcdf', max_pin='x.x.x') }} {{ mpi_prefix }}_*
+  ignore_run_exports_from:
+    - netcdf-fortran
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
+    - m4
+    - make
+  host:
+    - {{ mpi }}
+    # these need to be listed twice so conda build picks up the pins
+    - libnetcdf
```

Comment:
There is an overlinking warning on OSX for libnetcdf. Please check that you're not static linking.

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=662403&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310&l=7450

Reply:
Is this the issue holding up this PR?  The build link is no longer there, can you re trigger?  And if possible give more detail as to the issue - the configure line above clearly specifies shared and not static libraries.

Reply:
Here is the same but for mpich instead of openmpi:
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=722946&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310&l=7598

Reply:
@carterbox and @jedwards4b, even with `libnetcdf` 4.9.1, which should be built only as a dynamic library, I am seeing:
```
WARNING (libpnetcdf): run-exports library package conda-forge::libnetcdf-4.9.1-mpi_openmpi_haee62c1_3 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```
I really don't understand why.  Any advice on how to fix this would be appreciated.

Reply:
@jedwards4b, is there any way that my recipe could be misconfigured (for OSX in particular) and not be linking `libpnetcdf` with `libnetcdf` at all?

Reply:
Linking pnetcdf to netcdf is a new feature and one that I haven't used.  Do you need it?  What if we just remove the netcdf dependancy?

Reply:
I see, I assumed it was required.  We can remove it.

Reply:
What about `netcdf-fortran`.  I guess it can't need that either if it doesn't need `netcdf-c`?

Reply:
@jedwards4b, I think I'm starting to understand.  I have `--with-netcdf4=${PREFIX}` in the build and I will need to remove that if we're not using `libnetcdf`.  Otherwise, I see (predictably):
```
      ------------------------------------------------------------
      Missing NetCDF-4 header files 'netcdf.h' or 'netcdf_meta.h'
      required to build PnetCDF with NetCDF-4 support. Use option
          --with-netcdf4=/path/to/implementation
      to specify the location of NetCDF-4 installation. In addition,
      please make sure the MPI C compiler is compatible with the one
      used to build NetCDF-4. Check 'config.log' for more information.
      Stopping ...
      ------------------------------------------------------------
```
Would that be your suggestion to begin with?  Build without the `--with-netcdf4` flag?

Assuming that's the case, I'll try removing it.